mantidproject/mantid
# Add rebin string for q conversion in ISIS SANS GUI

~~**Do not merge or test before #20229 has been merged**~~

This PR does three things:
1. Adds an option to provide a rebin string for the q conversion in the ISIS SANS GUI
2. Moves the reduction dimensionality radio button to the `Run` tab
3. Provides `Thickness` as a option on the `Options` Column

Fixes #20268

**To test:**

Please find test data here: smb://olympic/babylon5/Scratch/Anton/GUI_TEST

1. Open the ISIS SANS v2 experimental interface
1. Select `SANS2D` as the instrument (below the data table)
2. For the user file load: USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_FRONT.txt
3. For the batch file load: batch_file.csv
4. Confirm that the reduction dimensionality radio buttons are on the front tab
5. On the first row of the data table in the `Options Column` start typing `Thickness`
    * Confirm it appears in the suggestions box
6. Add `Thickness=2` in the `Options Column
7. Go to the `Settings>StateDiagnostic` tab and press `Update row selection`
8. In the state tree go to `scale->thickness`. Confirm that it is set to 2
9. Switch the row to 1 and confirm that `scale->thickness` is not present.
10. Go to the `Settings>Q` tab.
11. On the `Q1D` line select `Variable` for the `Step type`
    * Confirm that the label of the first line edit field has changed to `Rebin String`. Confirm that the other line edit fields on that row have been disbaled.
12. Add "0.004,0.08,0.2, 0.06,1.0" in the line edit field
12. Go to the `State Diagnostic` tab, press `Update row selection` and check that `convert_to_q->q_1d_rebin_string` is "0.004,0.08,0.2, 0.06,1.0" 
13. Go to the front tab and press `Process`
    * Confirm that the reduction has no issues


**Release Notes** 
Not needed, since the GUI will be released only in this cycle

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
